### PR TITLE
Issue 4906 - Calibration skeleton pages

### LINKED ISSUE #4906: [2D MVP] Milestone 3: Calibration steps skeleton+routing
### Description
As groundwork we need a way to access the calibration pages so we can implement the functionality. This issue doesnt concern with design; just the route, showing a step progress (can be very crude) and a space in which the different steps of calibration are shown.

### Goals
- [ ] Users can access a route to calibrate their drawings.

### Tasks
- [x] For selecting the container/federation show a modal dialog displaying
- [x] Add a frontend route   ``v5/viewer/{teamspace}/{projectid}}/{containerId}/{revisionId}/calibration?drawing={drawingid}``
- [x] Add link from the drawing list to calibrate it.
- [x] Add mechanism (for now lets use context) for changing steps
- [x] Add Back/Cancel/Next buttons
- [x] The next button should only be enabled if the data in the content is valid
- [x] The back button should be enabled if it's not the first step
- [x] Cancel should always be enabled and go back to the page that triggered  the calibration wizard
- [x] Add select container/federation page skeleton
- [x] Add 3d point step page skeleton
- [x] Add 2d point step page skeleton
- [x] Add vertical spatial boundaries page skeleton page 
- [x] Add confirmation page skeleton page 

### Related Resources
[MockUp](https://www.figma.com/file/uTWnaYZ8IyoTn2i2xKtdBo/2D---3D?type=design&node-id=202-14332&mode=design&t=V5ZeaWfoqHoeg4bS-0)

### PR DESCRIPTION
**Pull request by Amantini1997**
This fixes #4906 

#### Description
Added the skeleton to navigate across the calibration steps

#### Test cases
Reach the calibration page click on the calibration button of any drawings (empty drawings should not lead there)

### CODE REVIEW

File: frontend/src/v5/ui/components/dashboard/dashboardProjectLayout/dashboardProjectDataFetcher.component.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { TeamspacesActionsDispatchers, ProjectsActionsDispatchers } from '@/v5/services/actionsDispatchers';
+import { DashboardParams } from '@/v5/ui/routes/routes.constants';
+
+export const DashboardProjectDataFetcher = () => {
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/components/dashboard/dashboardProjectLayout/dashboardProjectDataFetcher.component.tsx:24]: This look more like a hook than a component

File: frontend/src/v5/ui/components/dashboard/dashboardProjectLayout/dashboardProjectLayout.component.tsx [hunk 2]
```diff
@@ -15,42 +15,22 @@
...
-		<>
-			<AppBar />
-			<ProjectNavigation />
-			<Content>
-				{children}
-			</Content>
-		</>
-	);
-};
+export const DashboardProjectLayout = ({ children }: IDashboardProjectLayout): JSX.Element => (
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/components/dashboard/dashboardProjectLayout/dashboardProjectLayout.component.tsx:28]: lets restore this one to the previous state and create a calibration layout

File: frontend/src/v5/ui/routes/dashboard/projects/tickets/selectMenus/containersAndFederationsFormSelect.component.tsx [hunk 3]
```diff
@@ -15,21 +15,22 @@
...
 
-import { MultiSelectMenuItem } from '@controls/inputs/multiSelect/multiSelectMenuItem/multiSelectMenuItem.component';
-import { FormattedMessage } from 'react-intl';
-import { formatMessage } from '@/v5/services/intl';
 import { ContainersHooksSelectors, FederationsHooksSelectors } from '@/v5/services/selectorsHooks';
-import { FormSearchSelect } from '@controls/inputs/formInputs.component';
-import { sortByName } from '@/v5/store/store.helpers';
-import { ListSubheader } from '../tickets.styles';
 import { openUnsavedNewTicketWarningModal } from './selectMenus.helpers';
+import { ModelSelect, ModelSelectProps } from '@components/shared/modelSelect/modelSelect.component';
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/dashboard/projects/tickets/selectMenus/containersAndFederationsFormSelect.component.tsx:20]: why is this file being changed ? I dont think the relation with calibration skeleton

File: frontend/src/v5/ui/routes/dashboard/projects/calibration/calibration.component.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+				</EmptyPageView>
+			</Content>
+		);
+	}
+
+	return (
+		<CalibrationContextComponent>
+			<CalibrationHeader />
+			<Content>
+				<CalibrationStep />
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/dashboard/projects/calibration/calibration.component.tsx:91]: I dont see mockup content when the step changes.
image

The idea is that next dev can replace the content in the mockup components and make them functional

File: frontend/src/v5/ui/routes/dashboard/projects/calibration/calibrationSteps/calibrationStep.component.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+};
+
+export const CalibrationStep = () => {
+	const { step, setIsStepValid } = useContext(CalibrationContext);
+
+	useEffect(() => {
+		setIsStepValid(false);
+	}, [step]);
+
+	if (step === 0) return (<CalibrationStepContent />);
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/dashboard/projects/calibration/calibrationSteps/calibrationStep.component.tsx:34]: dont we have something similar to this in userSignupFormComponent? maybe there's a way to abstract it and make it reusable. 
Or at least inspire in that design.

**Review by sanmont3drepo — CHANGES_REQUESTED** [review]

File: frontend/src/v5/ui/components/dashboard/dashboardProjectLayout/dashboardProjectDataFetcher.component.tsx (hunk 1 above)
Comment (Amantini1997) [reply frontend/src/v5/ui/components/dashboard/dashboardProjectLayout/dashboardProjectDataFetcher.component.tsx:24]: Even though it doesn't return anything?

Comment (Amantini1997) [reply frontend/src/v5/ui/components/dashboard/dashboardProjectLayout/dashboardProjectDataFetcher.component.tsx:24]: Anyways, I'm probably ditching this design after the talk we had this morning where the NavigationBar can be passed a parameter

File: frontend/src/v5/ui/routes/dashboard/projects/calibration/calibrationStep/calibrationStep.component.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+							type={type}
+							id={type + '-panel-button'}
+							active={leftPanels.includes(type)}
+						/>
+					))}
+				</LeftPanelsButtons>
+				{step === 0 && <Calibration3DStep />}
+				{step === 1 && <Calibration2DStep />}
+				{step === 2 && <VerticalSpatialBoundariesStep />}
+				{step === 3 && <CalibrationConfirmationStep />}
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/dashboard/projects/calibration/calibrationStep/calibrationStep.component.tsx:63]: this logic would be inside the viewer to avoid reloading the model

**Review by sanmont3drepo — CHANGES_REQUESTED** [review]

File: frontend/src/v5/ui/routes/dashboard/projects/calibration/calibrationStep/calibrationStep.component.tsx (hunk 6 above)
Comment (Amantini1997) [reply frontend/src/v5/ui/routes/dashboard/projects/calibration/calibrationStep/calibrationStep.component.tsx:60]: I moved the whole calibration component inside the viewer, so, technically speaking, this is now inside the viewer. Did you mean something different though?

File: frontend/src/v5/ui/routes/dashboard/projects/calibration/calibration.component.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+								<FormattedMessage
+									id="calibration.invalidDrawing.goToDrawings"
+									defaultMessage="Open drawings"
+								/>
+							</Link>
+						</Button>
+					</EmptyPageView>
+				)
+			}
+		</Container>
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/dashboard/projects/calibration/calibration.component.tsx:55]: TBH Im not sure if this whole thing is necessary

File: frontend/src/v5/ui/routes/viewer/viewer.tsx [hunk 8]
```diff
@@ -91,20 +92,11 @@ export const Viewer = () => {
...
-	};
-
 	return (
-		<>
+		<CalibrationContextComponent>
 			<OpenTicketFromUrl />
 			<CheckLatestRevisionReadiness />
-			<ViewerGui match={v4Match} key={containerOrFederation} />
-		</>
+			<ViewerCanvases />
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/viewer/viewer.tsx:99]: Why did you change  where viewercancases is? is there's a good reason talk to Dan about it

**Review by sanmont3drepo — CHANGES_REQUESTED** [review]

File: frontend/src/v5/ui/routes/dashboard/projects/calibration/calibration.component.tsx (hunk 7 above)
Comment (Amantini1997) [reply frontend/src/v5/ui/routes/dashboard/projects/calibration/calibration.component.tsx:55]: What should happen if the drawingId is incorrect or that drawing no longer exists?

Comment (sanmont3drepo) [reply frontend/src/v5/ui/routes/dashboard/projects/calibration/calibration.component.tsx:55]: Nothing, like no calibration is there

File: frontend/src/v5/ui/routes/dashboard/projects/calibration/calibrationStep/calibration3DStep/calibration3DStep.component.tsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+import { ContainerRevisionsActionsDispatchers } from '@/v5/services/actionsDispatchers';
+import { modelIsFederation } from '@/v5/store/tickets/tickets.helpers';
+import { BasicStep } from '../basicStep.component';
+
+export const Calibration3DStep = () => {
+	const { containerOrFederation, teamspace, project } = useParams<ViewerParams>();
+	const isFed = modelIsFederation(containerOrFederation);
+
+	useEffect(() => {
+		if (!containerOrFederation || isFed) return;
```
[the remark this answers is not in the document]
Comment (Amantini1997) [reply frontend/src/v5/ui/routes/dashboard/projects/calibration/calibrationStep/calibration3DStep/calibration3DStep.component.tsx:29]: Left over from previous commits

File: frontend/src/v5/ui/routes/dashboard/projects/calibration/calibration.component.tsx (hunk 7 above)
Comment (sanmont3drepo) [reply frontend/src/v5/ui/routes/dashboard/projects/calibration/calibration.component.tsx:55]: In any case there is no design for this case, so maybe in the future we might want to change it. But for now, keep it simple.

File: frontend/src/v5/ui/routes/dashboard/projects/tickets/selectMenus/selectMenus.styles.ts [hunk 10]
```diff
@@ -14,9 +14,15 @@
  *  You should have received a copy of the GNU Affero General Public License
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-import ListSubheaderBase from '@mui/material/ListSubheader';
 import { MenuItem as MenuItemBase } from '@mui/material';
 import styled, { css } from 'styled-components';
+import ListSubheaderBase from '@mui/material/ListSubheader';
+
+export const MenuItem = styled(MenuItemBase)`
```
[the remark this answers is not in the document]
Comment (Amantini1997) [reply frontend/src/v5/ui/routes/dashboard/projects/tickets/selectMenus/selectMenus.styles.ts:21]: I moved this styled-component into a different file, and then brought it back here, but doing so, I place it in a different line. I will reposition it though, so it won't appear as a change

File: frontend/src/v5/ui/routes/viewer/viewer.tsx [hunk 11]
```diff
@@ -97,16 +100,16 @@ export const Viewer = () => {
 			model: containerOrFederation,
 			project,
 			teamspace,
-			revision,
+			revision: isFed ? undefined : revision,
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/viewer/viewer.tsx:103]: I think the check be 'revision === calibration' to be more explicit about this

**Review by sanmont3drepo — CHANGES_REQUESTED** [review]

File: frontend/src/v5/ui/routes/viewer/viewer.tsx (hunk 11 above)
Comment (Amantini1997) [reply frontend/src/v5/ui/routes/viewer/viewer.tsx:103]: But what if someone names their revision "calibration"?

Comment (sanmont3drepo) [reply frontend/src/v5/ui/routes/viewer/viewer.tsx:103]: that sucks, but fair enough. can you add a comment in the code to clarify the situation?

File: frontend/src/v5/ui/routes/viewer/viewer.tsx [hunk 12]
```diff
@@ -30,11 +29,15 @@ import { SpinnerLoader } from '@controls/spinnerLoader';
...
 import { ViewerCanvases } from '../dashboard/viewerCanvases/viewerCanvases.component';
+import { CalibrationContextComponent } from '../dashboard/projects/calibration/calibrationContext';
+import { ViewerGui } from '@/v4/routes/viewerGui';
+import { modelIsFederation } from '@/v5/store/tickets/tickets.helpers';
 
 export const Viewer = () => {
 	const [fetchPending, setFetchPending] = useState(true);
 
 	const { teamspace, containerOrFederation, project, revision } = useParams<ViewerParams>();
+	const isFed = modelIsFederation(containerOrFederation);
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/routes/viewer/viewer.tsx:40]: Does this check make sense now that we are using query strings for the calibration mode?

**Review by sanmont3drepo — APPROVED** [review]
